feat: add support of arbitrary mounting point via attachTo option

jnields opened the pull request:
This allows for users to specify where in the document their component should attach, either through
a CSS selector string or a provided HTMLElement. This option is passed through directly to the
vm.$mount method that is called as part of mount.js

feat #1487

This enables testing of SSR code with Vue test utils as well as enables rendering of applications via Vue test utils in contexts that aren't 100% Vue

<!--
Please make sure to read the Pull Request Guidelines:
https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [X] It's submitted to the `dev` branch.
- [X] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [X] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [X] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [X] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

jnields commented:
I added in a message that `attachToDocument` would be deprecated in the future. Let me know if you'd like me to remove it or update any messaging.

I also didn't add any checks to see if both `attachTo` and `attachToDocument` were provided. I can add such a check and messaging for that if you think it's necessary.

Also happy to help with any documentation updates that might be required. Please let me know.

Thanks!

lmiller1990 commented:
Hi! Thanks for the PR.

Seems reasonable. Happy to merge this... but can you please update the docs to include this? Otherwise no-one will be able to use this new feature!

lmiller1990 commented:
@jnields do you think you will have some time to update the docs by any chance? Going to do a 1.0 release pretty soon - would be good to get this in.

jnields commented:
@lmiller1990 awesome, thank you! I pushed up some doc updates in English.

jnields committed:
feat: add support of arbitrary mounting point via attachTo option

This allows for users to specify where in the document their component
should attach, either through a CSS selector string or a provided
HTMLElement. This option is passed through directly to the vm.$mount
method that is called as part of mount.js. This enables testing of SSR
code with Vue test utils as well as rendering of applications via
vue-test-utils in contexts that aren't entirely Vue

fixes #1492

lmiller1990 commented:
Looks good. This will go out with 1.0. We are finalizing the API. Aiming to have something out within a month.
